### PR TITLE
control/controlclient: fix accidental backoff reset

### PR DESCRIPTION
**Pull request by win-t**
fix for https://github.com/tailscale/tailscale/issues/7501

feel free to close this PR if the tailscale author's intention is actually to reset the backoff

### CODE REVIEW

**Comment by DentonGentry** [comment]
This PR looks correct to me, create a new err in order to make controlclient backoff.

**Comment by maisem** [comment]
> @win-t I don't think you want to reset here always. Does headscale always return the same url i.e. `goal.url == url`?

You could do this instead
```diff
diff --git a/control/controlclient/auto.go b/control/controlclient/auto.go
index 722288d9..d19b8700 100644
--- a/control/controlclient/auto.go
+++ b/control/controlclient/auto.go
@@ -407,7 +407,13 @@ func (c *Auto) authRoutine() {
 				c.mu.Unlock()
 
 				c.sendStatus("authRoutine-url", err, url, nil)
-				bo.BackOff(ctx, err)
+				if goal.url == url {
+					// The server sent us the same URL we already tried,
+					// backoff to avoid a busy loop.
+					bo.BackOff(ctx, errors.New("login URL not changing"))
+					continue
+				}
+				bo.BackOff(ctx, nil)
 				continue
 			}
 ```

**Comment by win-t** [comment]
@maisem, changed, based on your suggestion

File: control/controlclient/auto.go [hunk 1]
```diff
@@ -407,7 +407,13 @@ func (c *Auto) authRoutine() {
 				c.mu.Unlock()
 
 				c.sendStatus("authRoutine-url", err, url, nil)
-				bo.BackOff(ctx, err)
+				if goal.url == url {
+					// The server sent us the same URL we already tried,
+					// backoff to avoid a busy loop.
+					bo.BackOff(ctx, fmt.Errorf("login URL not changing"))
```
Comment (maisem) [inline control/controlclient/auto.go:413]: Don't do fmt.Errorf, there is no format directive here.

Use errors.New

Comment (win-t) [reply control/controlclient/auto.go:413]: okay

**Review by maisem — APPROVED** [review]

**Pull request merged**